### PR TITLE
Embedded objects fixes

### LINKED ISSUE #270: Item page issues
Bitstreams are no longer shown on some of the item pages.
The following error appears in the console when trying to visit an item linked from the search page:

`ERROR TypeError: bitstreams.filter is not a function`

The item's bitstreams are not shown or - in some cases - the whole item page stays blank except for a small button outline. 
Refreshing the page makes it load forever.

After a quick inspection, it seems there is something wrong with the way embeds (especially paginated embeds) are processed. 
I am working on a PR to fix this issue.

### PR DESCRIPTION
**Pull request by LotteHofstede**
This PR closes #270 
I rewrote the `BaseResponseParsingService` so it's able to parse (paginated) embedded objects as expected. Also made sure the `RemoteDataBuildService` knows how to handle paginated embeds. Added model classes and serialization mapping for `BitstreamFormat` and `ResourcePolicy`, so they can already be parsed from the responses where they are embedded.

### CODE REVIEW

File: src/app/core/shared/resource-policy.model.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+import { CacheableObject } from '../cache/object-cache.reducer';
+import { ResourceType } from './resource-type';
+
+export class ResourcePolicy implements CacheableObject {
```
Comment (tdonohue) [inline src/app/core/shared/resource-policy.model.ts]: This class should have comments to describe the various fields.  See, for example, `item.model.ts`

File: src/app/core/cache/models/normalized-resource-policy.model.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { autoserialize, autoserializeAs, inheritSerialization } from 'cerialize';
+import { ResourcePolicy } from '../../shared/resource-policy.model';
+
+import { mapsTo } from '../builders/build-decorators';
+import { IDToUUIDSerializer } from '../it-to-uuid-serializer';
+import { NormalizedObject } from './normalized-object.model';
+
+@mapsTo(ResourcePolicy)
+@inheritSerialization(NormalizedObject)
+export class NormalizedResourcePolicy extends NormalizedObject {
```
Comment (tdonohue) [inline src/app/core/cache/models/normalized-resource-policy.model.ts]: This class should have comments to describe it.  What is "normalized" about this ResourcePolicy object?

File: src/app/core/cache/models/normalized-bitstream-format.model.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { autoserialize, autoserializeAs, inheritSerialization } from 'cerialize';
+import { BitstreamFormat } from '../../shared/bitstream-format.model';
+
+import { mapsTo } from '../builders/build-decorators';
+import { IDToUUIDSerializer } from '../it-to-uuid-serializer';
+import { NormalizedObject } from './normalized-object.model';
+
+@mapsTo(BitstreamFormat)
+@inheritSerialization(NormalizedObject)
+export class NormalizedBitstreamFormat extends NormalizedObject {
```
Comment (tdonohue) [inline src/app/core/cache/models/normalized-bitstream-format.model.ts]: This Class should have comments to describe it & its properties. What is "normalized" about this BitstreamFormat object?

File: src/app/core/cache/it-to-uuid-serializer.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { hasValue } from '../../shared/empty.util';
+
+export class IDToUUIDSerializer {
```
Comment (tdonohue) [inline src/app/core/cache/it-to-uuid-serializer.ts]: Again, comments needed to describe class.

File: src/app/+item-page/field-components/collections/collections.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      fixture.detectChanges();
+    });
+
+    it('should show the collection', () => {
+      const collectionField = fixture.debugElement.query(By.css('ds-metadata-field-wrapper div.collections'));
+      expect(collectionField).not.toBeNull();
+    });
+  });
+
+  describe('When the requested item request has succeeded', () => {
```
Comment (tdonohue) [inline src/app/+item-page/field-components/collections/collections.component.spec.ts]: Should say "When the requested item request has failed".

**Comment by LotteHofstede** [comment]
Thanks for you feedback, Tim. I somehow missed it before.

I added documentation to all requested files and also extended or corrected some of the model classes. The test description you mentioned was also fixed, thanks for noticing!

The normalized model classes represent the objects as they are returned by the REST endpoint. In a lot of cases, objects have references to one or more objects, which are flattened into a UUIDs for serialization. This is why they are called normalized objects. 
The 'normal' model classes represent the version of the object as it is used by the UI.

**Pull request merged**